gear-programs/vft-treasury/src/services/msg_tracker.rs

```diff
@@ -0,0 +1,107 @@
+use super::Error;
+use gstd::{prelude::collections::HashMap, MessageId};
+use sails_rs::prelude::*;
+
+#[derive(Default, Debug)]
+pub struct MessageTracker {
+    pub message_info: HashMap<MessageId, MessageInfo>,
+}
+
+#[derive(Debug, Clone, Encode, Decode, TypeInfo)]
+pub struct MessageInfo {
+    pub status: MessageStatus,
+    pub details: TxDetails,
+}
+
+#[derive(Debug, Clone, Encode, Decode, TypeInfo)]
+pub enum TxDetails {
+    DepositVaraToTreasury {
```

Comment:
```suggestion
    DepositToTreasury {
```
There can be other tokens besides wVARA